src/binary/uint.rs

```diff
@@ -11,14 +11,14 @@ const MAX_UINT_SIZE_IN_BYTES: usize = mem::size_of::<UIntStorage>();
 /// [UInt and Int Fields](http://amzn.github.io/ion-docs/docs/binary.html#uint-and-int-fields)
 /// section of the binary Ion spec for more details.
 #[derive(Debug)]
-pub struct UInt {
+pub struct DecodedUInt {
     size_in_bytes: usize,
     value: UIntStorage,
```

Comment:
This is not your change, but I find the type alias really meaningless (it literally puzzled me until I went looking for it.  Maybe a question for @zslayton, but why bother with the alias here?

Reply:
When I first wrote this years ago, I vacillated between using a few types for storing magnitudes. `usize` was convenient because I could use the magnitude for various `std::collections` operations without casting. Then I decided I wanted it to be 8 bytes regardless of the platform, so I switched it to `u64`, which required refactoring a lot of API names in an era before CLion made that easy. THEN I heard rumblings that they were going to eventually add a `u128` primitive type (which they did [in Sept of 2020](https://github.com/rust-lang/rust/releases/tag/1.26.0)!) and wondered if I might some day think that was a better choice. I decided to use a type alias to simplify refactoring if/when I inevitably changed my mind again. :)

In short: we don't need it now. If we DO decide to change the storage type, refactoring is a lot easier in 2021.